appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java

```diff
@@ -733,10 +734,30 @@ private void readObject(ObjectInputStream in)
         // read values
         in.defaultReadObject();
 
+        if (repository != null) {
+            if (repository.isDirectory()) {
+                                // Check path for nulls
+                if (repository.getPath() == null || repository.getPath().contains("\0")) {
```

Comment:
If the repository.getPath() is really null, then it should not use the REPOSITORY_PATH_CONTAIN_NULL_CHARACTER message.